Services/Common/OJS.Services.Common.Data/Pagination/FilteringService.cs

```diff
@@ -110,6 +113,31 @@ public class FilteringService : IFilteringService
                         $"Property of type int cannot have {operatorType} operator");
             }
         }
+        else if (double.TryParse(value, out var doubleValue) && propertyType == typeof(double))
+        {
+            var constant = Expression.Constant(doubleValue, IsNullableType(property.Type) ? typeof(double?) : typeof(double));
+            switch (operatorType)
+            {
+                case OperatorType.Equals:
+                    expression = Expression.Equal(property, constant);
+                    break;
+                case OperatorType.GreaterThan:
+                    expression = Expression.GreaterThan(property, constant);
+                    break;
+                case OperatorType.LessThan:
+                    expression = Expression.LessThan(property, constant);
+                    break;
+                case OperatorType.LessThanOrEqual:
+                    expression = Expression.LessThanOrEqual(property, constant);
+                    break;
+                case OperatorType.GreaterThanOrEqual:
+                    expression = Expression.GreaterThanOrEqual(property, constant);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(
+                        $"Property of type int cannot have {operatorType} operator");
```

Comment:
This is repeated and could be extracted.

Reply:
Done . Also the operation for nullable types and the "null" value were extracted .